File: examples/_complete-min/app/lib/analytics.dart
// analytics.dart — data layer. Event names here must match spec/events.yaml exactly;
// trace.py checks this file against the spec for drift.
class AnalyticsEvent {
  static const appOpen = 'app_open';
  static const noteSaved = 'note_saved';
  static const reminderPermissionResult = 'reminder_permission_result';
}

void logEvent(String name, [Map<String, Object?> props = const {}]) {
  // MVP: local-only app, no network. Replace with local buffer + batched
  // upload if/when a backend is introduced (architecture.yaml: backend=none).
  assert(() {
    // ignore: avoid_print
    print('[analytics] $name $props');
    return true;
  }());
}

File: examples/web-complete-min/app/src/analytics.ts
// analytics.ts — data layer. Event names here must match spec/events.yaml exactly;
// trace.py checks this file against the spec for drift.

export const AnalyticsEvent = {
  bookmarkAdded: "bookmark_added",
  bookmarkDeleted: "bookmark_deleted",
  tagFilterApplied: "tag_filter_applied",
} as const;

export function logEvent(
  name: string,
  props: Record<string, unknown> = {},
): void {
  // MVP: local-only app, no network. Replace with a batched upload
  // if/when a backend is introduced (architecture.yaml: backend=none).
  if (
    typeof window !== "undefined" &&
    window.location.hostname === "localhost"
  ) {
    // eslint-disable-next-line no-console
    console.log("[analytics]", name, props);
  }
}
